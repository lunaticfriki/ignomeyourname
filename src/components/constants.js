// header
const headerTitle = 'i gnome your name!'
const headerSubtitle = 'Search your favorite gnome from Brastlewark!'

//nav
const navSearcher = 'Searcher'
const navAbout = 'About me'

//main
const mainTitle = ''
const mainubtitle = ''

// searcher
const sideSearcherTitle = 'SEARCHER'

//footer
const footerSignature = 'oscar ventura -'
const footerMessage = 'gnome searcher'

export {
  headerTitle,
  headerSubtitle,
  navSearcher,
  navAbout,
  mainTitle,
  mainubtitle,
  sideSearcherTitle,
  footerSignature,
  footerMessage
}
