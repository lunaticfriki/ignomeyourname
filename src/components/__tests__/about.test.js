import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import About from '../about'
import { translations } from '../constants'
import image from '../../assets/images/david-brooke-martin.jpg'

describe('About component suite', () => {
  const { aboutTitle, aboutArticle, aboutPhotoText } = translations
  const wrapperMount = mount(<About />)
  test('About snapshot', () => {
    const component = renderer.create(<About />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Should render a About component', () => {
    expect(wrapperMount).toBeTruthy()
  })
  test('Should render the texts', () => {
    expect(wrapperMount.find('h3').text()).toEqual(aboutTitle)
    expect(wrapperMount.find('article').text()).toEqual(aboutArticle)
    expect(wrapperMount.find('i').text()).toEqual(aboutPhotoText)
  })
  test('Should render an image', () => {
    expect(wrapperMount.find('img').prop('src')).toEqual(image)
  })
})
