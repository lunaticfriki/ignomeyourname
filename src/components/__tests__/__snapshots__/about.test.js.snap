// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component suite About snapshot 1`] = `
<div
  className="sc-AxjAm fxqOWP"
>
  <h3>
    Gnome me!
  </h3>
  <article>
    Hi, I'm Oscar Ventura, 34 years old and still searching the gender of all these gnomes...
  </article>
  <a
    href="https://unsplash.com/photos/50cGSHm1Jro"
    target="blank"
  >
    <img
      alt="gnome"
      src="david-brooke-martin.jpg"
    />
  </a>
  <i>
    Photo by David Brooke Martin on Unsplash
  </i>
</div>
`;
